packages/block-editor/src/components/block-list/block.js

```diff
@@ -301,8 +306,19 @@ const applyWithDispatch = withDispatch( ( dispatch, ownProps, { select } ) => {
 	// leaking new props to the public API (editor.BlockListBlock filter).
 	return {
 		setAttributes( newAttributes ) {
-			const { clientId } = ownProps;
-			updateBlockAttributes( clientId, newAttributes );
+			const {
+				clientId,
+				isFirstMultiSelected,
+				multiSelectedClientIds,
+			} = ownProps;
+
+			if ( isFirstMultiSelected ) {
+				multiSelectedClientIds.forEach( ( id ) => {
```

Comment:
It might be a good excuse to look into batching updates in some way. Unless `@wordpress/data` does it already with some magic :)

A naive (still reasonable) approach could be passing a list of ids as the first param to  `updateBlockAttributes` 🤷‍♂️

Reply:
I fixed this in 3b93437. I think allowing a list of ids is better than batching... Not sure how batching would work with undo/redo, but a single state change makes it one history record, which is what we want.